### PR TITLE
Add cpp-taskflow/2.2.0

### PR DESCRIPTION
**Pull request by Hopobcn**
Specify library name and version:  **cpp-taskflow/2.2.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/cpp-taskflow/all/test_package/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import os
+
+
+class TestPackageConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+
+    def configure(self):    
+        if not self.settings.compiler.cppstd:
+            self.settings.compiler.cppstd = 17
```
Comment (uilianries) [inline recipes/cpp-taskflow/all/test_package/conanfile.py]: do not force settings in your recipe.
```suggestion
```

File: recipes/cpp-taskflow/all/test_package/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+cmake_minimum_required(VERSION 2.8.12)
+project(test_package)
+
+set(CMAKE_VERBOSE_MAKEFILE TRUE)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
```
Comment (uilianries) [inline recipes/cpp-taskflow/all/test_package/CMakeLists.txt]:
```suggestion
target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
set_target_properties(${PROJECT_NAME} PROPERTIES CXX_STANDARD 17)
```

**Comment by uilianries** [comment]
Seems like apple-clang 9.1 has no support for *optional*

**Comment by uilianries** [comment]
@Hopobcn According cppreference, optional is only supported since apple-clang 10

https://en.cppreference.com/w/cpp/compiler_support

**Comment by Hopobcn** [comment]
This recipy will also benefit from https://github.com/conan-io/conan/pull/5997

**Pull request merged**